### ISSUE TITLE
is this a leak problem? 

### ISSUE THREAD

**Issue by xing-ou** [socketio/socket.io-client-swift#276]
i noticed a leak problem when i use the instrument ,the code is simple enough,it just new a socket.here is all the code  
```swift
import UIKit
import Socket_IO_Client_Swift
class ViewController: UIViewController {

    let socket = SocketIOClient(socketURL: "xxxxxx",options: [.Log(true), .ForcePolling(true)])
    override func viewDidLoad() {
        super.viewDidLoad()
        socket.on("connect") {data, ack in
            print("socket connected========")
        }
        socket.connect()
    }

    deinit{
        socket.close()
    }
}
```
and here is the instrument graph:
![leaksnapshot](https://cloud.githubusercontent.com/assets/13159955/12284010/2b1fec86-b9e7-11e5-9129-5c5e226d1cdb.png)

because i'm new in iOS ,so i don't know how to give more info for the tracking, but the code is simple enough,everyone can try it 
the Socket_IO_Client_Swift version is 4.1.6

**Comment by nuclearace** [comment]
See if it's fixed in 5.0.0

**Comment by xing-ou** [comment]
nice, it's fixed in 5.0.0
